README
NAME
    Dist::Zilla::PluginBundle::Author::DBR - DBRs Dist::Zilla PluginBundle

VERSION
    version 1.2.4

SYNOPSIS
    This PluginBundle is roughly equivalent to the following "dist.ini":

        # dist.ini
        [@Filter]
        -bundle          = @Classic
        -remove          = ExtraTests
        -remove          = PodVersion

        [Authority]
        authority        = cpan:DBR

        [AutoMetaResources]
        homepage            = https://metacpan.org/module/%{dist}
        repository.type     = git
        repository.url      = git://github.com/xdbr/%{dist}.git
        repository.web      = http://github.com/xdbr/%{dist}
        bugtracker.web      = http://github.com/xdbr/%{dist}/issues

        GithubMeta

        [@Git]
        changelog        = Changes             ; this is the default
        allow_dirty      = dist.ini            ; see Git::Check...
        allow_dirty      = Changes             ; ... and Git::Commit
        commit_msg       = v%v%n%n%c           ; see Git::Commit
        tag_format       = %v                  ; see Git::Tag
        tag_message      = %v                  ; see Git::Tag
        push_to          = origin              ; see Git::Push

        [Test::Portability]
        test_vms_length  = 0
        test_ansi_chars  = 0
        test_one_dot     = 0

        [PruneCruft]
        except           = '.gitignore'

        [ExecDir]
        dir              = bin

        [InstallRelease]
        install_command  = cpanm .

        [AutoPrereqs]
        [CheckChangesHasContent]
        [CheckExtraTests]
        [CheckPrereqsIndexed]
        [CheckVersionIncrement]
        [ConfirmRelease]
        [EOLTests]
        [HasVersionTests]
        [InstallGuide]
        [MetaJSON]
        [MetaProvides::Class]
        [MetaTests]
        [ModuleBuild]
        [NextRelease]
        [NoTabsTests]
        [PodWeaver]
        [ReadmeFromPod]
        [ReportVersions]
        [RunExtraTests]
        [SchwartzRatio]
        [ShareDir::Tarball]
        [SpellingCommonMistakesTests]
        [Test::ChangesHasContent]
        [Test::CheckDeps]
        [Test::CheckManifest]
        [Test::MinimumVersion]
        [Test::ReportPrereqs]
        [Test::UseAllModules]
        [TestRelease]


        [Prereqs]
        Dist::Zilla                                       = 0
        Dist::Zilla::Plugin::Authority                    = 0
        Dist::Zilla::Plugin::AutoMetaResources            = 0
        Dist::Zilla::Plugin::AutoPrereqs                  = 0
        Dist::Zilla::Plugin::CheckChangeLog               = 0
        Dist::Zilla::Plugin::CheckChangesHasContent       = 0
        Dist::Zilla::Plugin::CheckExtraTests              = 0
        Dist::Zilla::Plugin::CheckPrereqsIndexed          = 0
        Dist::Zilla::Plugin::CheckVersionIncrement        = 0
        Dist::Zilla::Plugin::ConfirmRelease               = 0
        Dist::Zilla::Plugin::EOLTests                     = 0
        Dist::Zilla::Plugin::HasVersionTests              = 0
        Dist::Zilla::Plugin::InstallGuide                 = 0
        Dist::Zilla::Plugin::InstallRelease               = 0
        Dist::Zilla::Plugin::MetaJSON                     = 0
        Dist::Zilla::Plugin::MetaProvides::Class          = 0
        Dist::Zilla::Plugin::MetaProvides::Package        = 0
        Dist::Zilla::Plugin::MetaTests                    = 0
        Dist::Zilla::Plugin::MinimumPerl                  = 0
        Dist::Zilla::Plugin::ModuleBuild                  = 0
        Dist::Zilla::Plugin::NextRelease                  = 0
        Dist::Zilla::Plugin::NoTabsTests                  = 0
        Dist::Zilla::Plugin::PkgVersion                   = 0
        Dist::Zilla::Plugin::PodCoverageTests             = 0
        Dist::Zilla::Plugin::PodSyntaxTests               = 0
        Dist::Zilla::Plugin::PodWeaver                    = 0
        Dist::Zilla::Plugin::ReadmeFromPod                = 0
        Dist::Zilla::Plugin::ReportPhase                  = 0
        Dist::Zilla::Plugin::ReportVersions               = 0
        Dist::Zilla::Plugin::Run::Release                 = 0
        Dist::Zilla::Plugin::RunExtraTests                = 0
        Dist::Zilla::Plugin::SchwartzRatio                = 0
        Dist::Zilla::Plugin::ShareDir::Tarball            = 0
        Dist::Zilla::Plugin::SpellingCommonMistakesTests  = 0
        Dist::Zilla::Plugin::TaskWeaver                   = 0
        Dist::Zilla::Plugin::Test::ChangesHasContent      = 0
        Dist::Zilla::Plugin::Test::CheckDeps              = 0
        Dist::Zilla::Plugin::Test::CheckManifest          = 0
        Dist::Zilla::Plugin::Test::Compile                = 0
        Dist::Zilla::Plugin::Test::Legal                  = 0
        Dist::Zilla::Plugin::Test::MinimumVersion         = 0
        Dist::Zilla::Plugin::Test::Perl::Critic           = 0
        Dist::Zilla::Plugin::Test::Portability            = 0
        Dist::Zilla::Plugin::Test::ReportPrereqs          = 0
        Dist::Zilla::Plugin::Test::UseAllModules          = 0
        Dist::Zilla::Plugin::TestRelease                  = 0
        Dist::Zilla::PluginBundle::Git                    = 0
        Dist::Zilla::Role::PluginBundle::Merged           = 0
        Pod::Weaver::Plugin::WikiDoc                      = 0
        Test::CPAN::Meta                                  = 0
        Test::UseAllModules                               = 0

AUTHOR
    Daniel B. <dbr@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2012 by Daniel B..

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.

SUPPORT
  Websites
    The following websites have more information about this module, and may
    be of help to you. As always, in addition to those websites please use
    your favorite search engine to discover more resources.

    *   MetaCPAN

        A modern, open-source CPAN search engine, useful to view POD in HTML
        format.

        <http://metacpan.org/release/Dist-Zilla-PluginBundle-Author-DBR>

  Email
    You can email the author of this module at "DBR@cpan.org" asking for
    help with any problems you have.

  Bugs / Feature Requests
    Please report any bugs or feature requests through the web interface at
    <https://github.com/xdbr/Dist-Zilla-PluginBundle-Author-DBR/issues>. You
    will be automatically notified of any progress on the request by the
    system.

  Source Code
    The code is open to the world, and available for you to hack on. Please
    feel free to browse it and play with it, or whatever. If you want to
    contribute patches, please send me a diff or prod me to pull from your
    repository :)

    <https://github.com/xdbr/Dist-Zilla-PluginBundle-Author-DBR>

      git clone https://github.com/xdbr/Dist-Zilla-PluginBundle-Author-DBR.git



Changes
1.2.4     2014-01-07 17:05:08 Europe/Berlin

  - Add DZ::Plugin::MetaYAML

1.2.3     2014-01-04 11:29:55 Europe/Berlin
  
  - Merge pull request #1 from sergeyromanov/patch-1

1.2.2     2013-09-25 21:46:02 Europe/Berlin

    - Still trouble with AutoMetaResources (remove)...
    - Add Pod::Weaver::Section::Support

1.2.0     2013-09-25 21:17:06 Europe/Berlin

    - Add GithubMeta

1.1.0     2013-09-25 21:10:26 Europe/Berlin

    - Let's try AutoMetaResources on this module itself

1.0.0     2013-09-25 20:41:52 Europe/Berlin

    - Remove [AutoMetaResources] again, seems to not work in vanilla dist.
    - Let's just try again with AutoMetaResources
    - Change to semantic versioning
    - add Dist::Zilla::Test::CPAN::Changes

0.930     2013-07-04 08:59:41 Europe/Berlin

    - Add more information to [AutoMetaResources]

0.920     2013-07-04 00:30:45 Europe/Berlin

    - Add Documentation

0.900     2013-07-04 00:23:54 Europe/Berlin

    - Finally getting Dist::Zilla::Role::PluginBundle::Merged under control...

0.842     2013-07-03 23:20:21 Europe/Berlin

    - Adding Changes stuff.

0.840

    - This was the beginning, back in the days, when no Changelog was kept.

Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "DBRs Dist::Zilla PluginBundle",
  "dist_author" => [
    "Daniel B. <dbr\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-PluginBundle-Author-DBR",
  "dist_version" => "1.2.4",
  "license" => "perl",
  "module_name" => "Dist::Zilla::PluginBundle::Author::DBR",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla" => 0,
    "Dist::Zilla::Plugin::Authority" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::CheckChangeLog" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::CheckVersionIncrement" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::EOLTests" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => 0,
    "Dist::Zilla::Plugin::HasVersionTests" => 0,
    "Dist::Zilla::Plugin::InstallGuide" => 0,
    "Dist::Zilla::Plugin::InstallRelease" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Class" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
    "Dist::Zilla::Plugin::MetaTests" => 0,
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::ModuleBuild" => 0,
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::NoTabsTests" => 0,
    "Dist::Zilla::Plugin::PkgVersion" => 0,
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::ReadmeFromPod" => 0,
    "Dist::Zilla::Plugin::ReportPhase" => 0,
    "Dist::Zilla::Plugin::ReportVersions" => 0,
    "Dist::Zilla::Plugin::Run::Release" => 0,
    "Dist::Zilla::Plugin::RunExtraTests" => 0,
    "Dist::Zilla::Plugin::SchwartzRatio" => 0,
    "Dist::Zilla::Plugin::ShareDir::Tarball" => 0,
    "Dist::Zilla::Plugin::SpellingCommonMistakesTests" => 0,
    "Dist::Zilla::Plugin::TaskWeaver" => 0,
    "Dist::Zilla::Plugin::Test::CPAN::Changes" => 0,
    "Dist::Zilla::Plugin::Test::ChangesHasContent" => 0,
    "Dist::Zilla::Plugin::Test::CheckDeps" => 0,
    "Dist::Zilla::Plugin::Test::CheckManifest" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Legal" => 0,
    "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
    "Dist::Zilla::Plugin::Test::UseAllModules" => 0,
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::PluginBundle::Git" => 0,
    "Dist::Zilla::Role::PluginBundle::Merged" => 0,
    "MooseX::Declare" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::Section::Support" => 0,
    "Test::CPAN::Meta" => 0,
    "Test::UseAllModules" => 0,
    "true" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Carp" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Scalar::Util" => 0,
    "Test::CheckDeps" => "0.006",
    "Test::More" => "0.94",
    "Test::UseAllModules" => 0,
    "perl" => "5.004",
    "strict" => 0,
    "warnings" => 0
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "DBRs Dist::Zilla PluginBundle",
  "AUTHOR" => "Daniel B. <dbr\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Author-DBR",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::PluginBundle::Author::DBR",
  "PREREQ_PM" => {
    "Dist::Zilla" => 0,
    "Dist::Zilla::Plugin::Authority" => 0,
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::CheckChangeLog" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::CheckVersionIncrement" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::EOLTests" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => 0,
    "Dist::Zilla::Plugin::HasVersionTests" => 0,
    "Dist::Zilla::Plugin::InstallGuide" => 0,
    "Dist::Zilla::Plugin::InstallRelease" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Class" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
    "Dist::Zilla::Plugin::MetaTests" => 0,
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::ModuleBuild" => 0,
    "Dist::Zilla::Plugin::NextRelease" => 0,
    "Dist::Zilla::Plugin::NoTabsTests" => 0,
    "Dist::Zilla::Plugin::PkgVersion" => 0,
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::ReadmeFromPod" => 0,
    "Dist::Zilla::Plugin::ReportPhase" => 0,
    "Dist::Zilla::Plugin::ReportVersions" => 0,
    "Dist::Zilla::Plugin::Run::Release" => 0,
    "Dist::Zilla::Plugin::RunExtraTests" => 0,
    "Dist::Zilla::Plugin::SchwartzRatio" => 0,
    "Dist::Zilla::Plugin::ShareDir::Tarball" => 0,
    "Dist::Zilla::Plugin::SpellingCommonMistakesTests" => 0,
    "Dist::Zilla::Plugin::TaskWeaver" => 0,
    "Dist::Zilla::Plugin::Test::CPAN::Changes" => 0,
    "Dist::Zilla::Plugin::Test::ChangesHasContent" => 0,
    "Dist::Zilla::Plugin::Test::CheckDeps" => 0,
    "Dist::Zilla::Plugin::Test::CheckManifest" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Legal" => 0,
    "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
    "Dist::Zilla::Plugin::Test::UseAllModules" => 0,
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::PluginBundle::Git" => 0,
    "Dist::Zilla::Role::PluginBundle::Merged" => 0,
    "MooseX::Declare" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::Section::Support" => 0,
    "Test::CPAN::Meta" => 0,
    "Test::UseAllModules" => 0,
    "true" => 0
  },
  "TEST_REQUIRES" => {
    "Carp" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Scalar::Util" => 0,
    "Test::CheckDeps" => "0.006",
    "Test::More" => "0.94",
    "Test::UseAllModules" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "1.2.4",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



